homeassistant/components/guardian/__init__.py

```diff
@@ -164,25 +389,14 @@ async def _async_internal_added_to_hass(self):
         """
         raise NotImplementedError
 
-    @callback
-    def _async_update_from_latest_data(self):
-        """Update the entity.
-
-        This should be extended by Guardian platforms.
-        """
-        raise NotImplementedError
-
     @callback
     def async_add_coordinator_update_listener(self, api: str) -> None:
         """Add a listener to a DataUpdateCoordinator based on the API referenced."""
-
-        @callback
-        def async_update():
-            """Update the entity's state."""
-            self._async_update_from_latest_data()
-            self.async_write_ha_state()
-
-        self.async_on_remove(self._coordinators[api].async_add_listener(async_update))
+        self.async_on_remove(
+            self._coordinators[api].async_add_listener(
+                self._async_update_state_callback
+            )
+        )
 
     async def async_added_to_hass(self) -> None:
         """Perform tasks when the entity is added."""
```

Comment:
Wow, why are you calling internal added to hass? That's internal…

Reply:
I was majorly confused, but I see what happened: I used this method name, totally unaware that it was already used in `Entity`. 😂 I'll change the name.